### ISSUE TITLE
Editing in files that are blocked because of being in the dependency chain does not automatically unblock them

### ISSUE THREAD

**Issue by proofbot** [ProofGeneral/PG#311]
Note: the issue was imported automatically using json2github.py

Original issue: psteckler/ProofGeneral#75
Opened by: @RalfJung

Say file `B` imports `A`, and I have the interactive focus in `B`. If I now open `A`, PG shows the entire file as red because things are "locked", which makes sense -- after all, changes here could break `B`. However, when I try to type in that red region, PG master automatically offers to change the interactive focus to `A` -- while this branch of PG just refuses to change the text. I have to manually C-c C-RET, confirm, C-c C-RET. It would be nice if the behavior from PG master could be adapted here (actually I am not even sure why it is different, but I guess this is not part of the code that can be shared).
